src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/BeanstalkInstanceTypeValidator.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.Data;
+
+namespace AWS.Deploy.Common.Recipes.Validation;
+
+public class BeanstalkInstanceTypeValidator : IRecipeValidator
+{
+    private readonly IOptionSettingHandler _optionSettingHandler;
+    private readonly IAWSResourceQueryer _awsResourceQueryer;
+
+    public BeanstalkInstanceTypeValidator(IAWSResourceQueryer awsResourceQueryer, IOptionSettingHandler optionSettingHandler)
+    {
+        _awsResourceQueryer = awsResourceQueryer;
+        _optionSettingHandler = optionSettingHandler;
+    }
+
+    public string InstanceTypeOptionSettingsId { get; set; } = "InstanceType";
+    public string ApplicationNameOptionSettingsId { get; set; } = "BeanstalkApplication.ApplicationName";
+    public string EnvironmentNameOptionSettingsId { get; set; } = "BeanstalkEnvironment.EnvironmentName";
+
+    public async Task<ValidationResult> Validate(Recommendation recommendation, IDeployToolValidationContext deployValidationContext)

```

Comment:
can we split `Validate` into multiple sub functions - it is pretty long right now

Reply:
I didn't split the function as I am returning validation results as well as retrieving values at every step and didn't want to overengineer it in order to make the split. However, I did add comments at every step to explain the logic.